### ISSUE TITLE
LED API not taking advantage of 13-bit ESP32 capability

### ISSUE THREAD

**Issue by nathancheek** [HomeSpan/HomeSpan#115]
Currently, `LedPin::set()` takes an integer percentage, meaning there are only 101 duty cycle steps for each LED: `0-100`, not even 7 bits of precision. At the configured 5kHz frequency, the ESP32 supports 13-bit precision for duty cycle, i.e. 8192 steps. I propose that the library support all 13 bits.

The following line could be changed to use `LEDC_TIMER_13_BIT`.
https://github.com/HomeSpan/HomeSpan/blob/69efec0d6545d6a920d79628edc6847f180d63f0/src/extras/PwmPin.cpp#L18

The `LedPin::LedPin(uint8_t pin, uint8_t level)` and `LedPin::LedPin(uint8_t pin, uint8_t level)` functions would also need to be modified to take appropriate duty cycle values for `level`. Since the library already processes lightbulb commands in floating point, it might make since to use a floating point value `0-100` for these function headers.

**Comment by HomeSpan** [comment]
Unfortunately, the HAP Brightness Characteristic only takes an INT, so any additional precision would not be accessible via HomeKit.

**Comment by nathancheek** [comment]
Yes, brightness is an int (0-100), but different combinations of hue and saturation could take advantage of further precision.

* Brightness levels: 101 (0-100)
* Hue levels: 360 (0-359)
* Saturation levels: 101 (0-100)

The mapping of HSV to RGB might have some shared outputs, so it might not be as much as 101 * 360 * 101 possible values. But it's definitely well above 100 values.

For example, if you set the brightness to 50%, and hue to 0º (red), HAP still supports 101 different saturation levels between red and white. Here are some examples below:

| Saturation | Red | Green | Blue |
| ---------- | --- | ----- | ---- |
| 0 | 0.5 | 0.5 | 0.5 |
| 0.25 | 0.5 | 0.375 |  0.375 |
| 0.5 | 0.5 | 0.25 |  0.25 |
| 0.75 | 0.5 | 0.125 |  0.125 |
| 1 | 0.5 | 0 | 0 |

Since the values for Red, Green, and Blue currently get converted to 1%-stepped integers, 37.5% cannot be produced. Increasing the precision would allow such values to be produced.

**Comment by HomeSpan** [comment]
That's a good point I had not thought of.  I should be able to make this update in the next release.  Thanks for the suggestion.

**Comment by HomeSpan** [comment]
HomeSpan 1.4.0 has just been released and includes new PWM functionality to allow you to set the PWM frequency (default is 5 kHz if not specified).  Resolution is automatically set to maximum possible for the chosen frequency.  In addition, the set() method now takes a 32-bit floating value instead of an int when setting the duty.  See [LedPin()](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Extras.md#ledpinuint8_t-pin-float-level-uint16_t-frequency) for details.
